modules/porous_flow/doc/content/modules/porous_flow/natural_convection.md

```diff
@@ -0,0 +1,94 @@
+
+!listing modules/porous_flow/examples/natural_convection/natural_convection.i block=Materials
+
+### Initial and Boundary Conditions
+
+The next step is to supply the initial and boundary conditions. For this problem, two initial conditions need to be implemented. The first one is a constant temperature of 285 K which was declared in the previous section. The second one is the hydrostatic variation of porepressure along the y direction as shown below:
+
+!listing modules/porous_flow/examples/natural_convection/natural_convection.i block=ICs
+
+Shifting to the boundary conditions. There are three Dirichlet boundary conditions that we need to supply. The first two are constant temperature at the top and heater surfaces denoted as "t_top" and "t_bot". And, the last one is constant pressure at the top surface denoted as "p_top".
+
+!listing modules/porous_flow/examples/natural_convection/natural_convection.i block=BCs
+
+### Executioner setup
+
+Lastly, we will tell Moose the type of the problem and how we want it to perform or so call Executioner setup (shown in the code block below). For this problem, a transient solver is required.
+To save time and computational resources, "IterationAdaptiveDT" and "Adaptivity"
+were implemented. The former enables the time step to be increased if the solution converges and decreased if it does not. This will help save time if a large time step can be used and aid in convergence with a small time step is required [(Details can be found here)](https://mooseframework.inl.gov/source/timesteppers/IterationAdaptiveDT.html). Similarly, the latter enables the mesh to automatically adapt to optimize the computational resource usage and resolve the area with stiff gradients. The mesh will be coarsened at the regions with low error and refined at the critical areas (high error and stiff gradients) [(Details can be found here)](https://mooseframework.inl.gov/moose/syntax/Adaptivity/). For practice, users can try to disable them by putting the code block into comment and witnessing the difference in the solving time and solution.
+
+!listing modules/porous_flow/examples/natural_convection/natural_convection.i block=Executioner
+
+### Auxilary kernels and variables (optional)
+
+This section is optional since these kernels and variables do not affect the calculation of the desired variable. Instead, they were declared as part of learning and for completeness since in practice, it is uncommon that we only care about a single property. In this example, we want to look at the fluid density at the end of the simulation thus we will add an auxkernel and an auxvariable for it. It can be noticed that the term "execute_on = TIMESTEP_END" indicates this kernel will be activated to calculate the fluid density at the last time step thus this calculation does not affect the performance of other tasks.
```

Comment:
```suggestion
This section is optional since these kernels and variables do not affect the calculation of the desired variable. In this example, we want to output the fluid density at the end of the simulation thus we will add an auxkernel and an auxvariable for it. It can be noticed that the term `execute_on = TIMESTEP_END` indicates this kernel will be activated to calculate the fluid density at the end of each time step only, thus this calculation does not affect the performance of other tasks.
```